### PR TITLE
Make 'building packages', clean up 'intro'

### PR DESCRIPTION
**Pull request by electronwill**
Make the 'building packages' folder and its content, by pulling things
out of the 'intro' file. Also create the Python2/3 environments page.

### CODE REVIEW

**Comment by electronwill** [comment]
OK, this is ready to go, and we can close PR 91. @asmeurer @kerrywatson1

File: docs/source/building/recipe.rst [hunk 1]
```diff
@@ -0,0 +1,673 @@
...
+------------------
+
+
+All the metadata in the recipe is specified in the ``meta.yaml`` file. All
+sections are optional except for package/name and package/version.
+
+.. code-block:: yaml
+
+
+    package:
```
Comment (asmeurer) [inline docs/source/building/recipe.rst]: We should no longer have this giant pseudo-meta.yaml. If you want, put a real meta.yaml here (like one of the ones from the build tutorials).

Comment (asmeurer) [inline docs/source/building/recipe.rst]: Oh, I guess this pull request doesn't split this out into sections yet. So never mind.

File: docs/source/building/meta-pkg.rst [hunk 2]
```diff
@@ -0,0 +1,8 @@
+-------------
+Meta-Packages
```
Comment (asmeurer) [inline docs/source/building/meta-pkg.rst]: This should be spelled "metapackage" (with no hyphen).

**Comment by asmeurer** [comment]
We need to figure out how to not break links here (particularly to build.html, and links to sections on that page if possible).

**Comment by asmeurer** [comment]
I think to do redirects, unless you can find some way to do it within Sphinx, will be to add a literal html file with something like

```html
<html>
<head>
   <meta http-equiv="refresh" content="0;URL=builder/recipe.html">
</head>
</html>
```

(and we need to see if that maintains anchors).

**Comment by asmeurer** [comment]
Other than that this looks good.

**Comment by electronwill** [comment]
It's slightly inelegant, but I went ahead and restored build.html to its original place, and arranged the navigation around it, so now everything will look the way it does in Kerry's plan, and all the old links will also still work. Once all these navigation changes are done we could explore using html redirects instead, but honestly, I'm not sure it would have any advantage over the current approach. For now, I think this pull request is ready to merge.

**Comment by kerrywatson1** [comment]
Great job, Will!

**Pull request merged**